### PR TITLE
Add loading screen image generation for pre-match notifications

### PR DESCRIPTION
**Pull request by shepherdjerred**
## Summary
This PR adds comprehensive loading screen image generation for pre-match notifications. When a tracked player enters a League of Legends game, the bot now generates a visual loading screen showing all 10 participants with their champions, runes, summoner spells, and ranked information, then sends it as an attachment in the Discord notification.

ranked-flex-5v5

### Data Models & Schemas
- **New `LoadingScreenData` model** (`packages/data/src/model/loading-screen.ts`): Defines the complete structure for loading screen information including participants, bans, queue info, and layout mode
- **Layout modes**: Supports "standard" (5v5), "aram" (5v5 no bans), and "arena" (8 teams of 2)
- **Comprehensive test coverage** for all loading screen schemas with edge cases

### Backend - Data Collection
- **`loading-screen-builder.ts`**: Orchestrates data collection from spectator API, resolves champion/skin info, fetches ranks for all 10 participants in parallel
- **`champion-resolver.ts`**: Maps Riot champion IDs to Data Dragon keys (e.g., 64 → "LeeSin")
- **`skin-resolver.ts`**: Extracts skin numbers from spectator API customization objects with fallback to default skin
- **`rank.ts`**: New `getRankByPuuid()` function to fetch solo queue rank for any player by PUUID and region

### Frontend - Image Rendering
- **`player-card.tsx`**: Individual player card component showing champion splash art, summoner name, rank badge, runes, and summoner spells (280x480px)
- **`game-header.tsx`**: Header displaying queue type, ranked badge, map name, and banned champions with red X overlay
- **`standard-layout.tsx`**: Two-column layout for 5v5 games (blue team vs red team)
- **`arena-layout.tsx`**: 4x2 grid layout for Arena mode (8 teams of 2)
- **`loading-screen.tsx`**: Main component that routes to appropriate layout and handles SVG/PNG generation
- **`index.tsx`**: Orchestrates image preloading and conversion to PNG via Satori + sharp

### Data Dragon Assets
- **Champion loading screen images**: Downloads tall portrait art for all champion skins from Data Dragon
- **`champion-skins.ts`**: Lookup module for available skin numbers per champion
- **`champion-skins.json`**: Generated mapping file of champions to valid skin numbers (excludes chromas)
- **Image cache**: Extended to support champion loading screen art with fallback to default skin

### Storage & Notifications
- **`s3-prematch.ts`**: New S3 storage module for prematch assets with configurable metadata and logging
- **`s3.ts`**: New functions to save spectator data (JSON), loading screen PNG, and SVG to S3
- **`prematch-notification.ts`**: Integrated loading screen generation with graceful degradation (sends text-only if image generation fails), fire-and-forget S3 saves, and metrics collection

### Metrics
- **`prematchLoadingScreenGeneratedTotal`**: Counter tracking successful/failed loading screen generations by queue type
- **`prematchLoadingScreenDurationSeconds`**: Histogram measuring generation time

## Notable Implementation Details

- **Graceful degradation**: If loading screen generation fails, the notification still sends with text-only content
- **Parallel rank fetching**: All 10 participant ranks are fetched concurrently for performance
- **Image preloading**: Champion loading screen art and ban icons are preloaded before rendering to avoid runtime lookups
- **Fallback handling**: If a specific skin isn't available, the cache automatically falls back to the default skin
- **Fire-and-forget S3 saves**: Spectator data, PNG, and SVG are saved to S3 asynchronously without blocking the Discord notification
- **Comprehensive test coverage**: Unit tests for schema validation, champion/skin resolution, and skin extraction from API responses

https://claude.ai/code/session_011LkMpiEvdUmwRLnZ7Vnxi5

### CODE REVIEW

File: packages/scout-for-lol/packages/backend/src/league/model/rank.ts [hunk 1]
```diff
@@ -61,6 +62,49 @@ export function getRank(
...
+    return getRank(parseResult.data, solo);
+  } catch (error) {
+    const status =
+      error instanceof Error && error.message.includes("timed out")
+        ? "timeout"
+        : "error";
+    riotApiRequestsTotal.inc({ source: "rank-by-puuid", status });
+    // Don't log at error level — rank fetch failures for non-tracked players are expected
+    logger.debug(`Failed to fetch rank for puuid ${puuid}: ${String(error)}`);
+    return undefined;
```
Comment (shepherdjerred) [inline packages/scout-for-lol/packages/backend/src/league/model/rank.ts]: why would this be expected?

File: packages/scout-for-lol/packages/backend/src/league/tasks/prematch/loading-screen-builder.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  trackedPuuids: Set<string>,
+  region: Region,
+): Promise<LoadingScreenData> {
+  const queueType = parseQueueType(gameInfo.gameQueueConfigId);
+  const queueDisplayName = queueType
+    ? queueTypeToDisplayString(queueType)
+    : gameInfo.gameMode;
+  const isRanked =
+    gameInfo.gameQueueConfigId === 420 ||
+    gameInfo.gameQueueConfigId === 440;
```
Comment (shepherdjerred) [inline packages/scout-for-lol/packages/backend/src/league/tasks/prematch/loading-screen-builder.ts]: do we have a shared utility? if not we prob should

File: packages/scout-for-lol/packages/data/src/model/loading-screen.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  spell2Id: z.number().int(),
+  /** Keystone rune ID (first perk in primary tree) */
+  keystoneRuneId: z.number().int().optional(),
+  /** Secondary rune tree ID */
+  secondaryTreeId: z.number().int().optional(),
+  /** Solo queue rank (fetched via LeagueV4) */
+  rank: RankSchema.optional(),
+  /** Whether this player is tracked by the bot */
+  isTrackedPlayer: z.boolean(),
+});
```
Comment (shepherdjerred) [inline packages/scout-for-lol/packages/data/src/model/loading-screen.ts]: these types can be tighter?
